src/realm/object-store/c_api/sync.cpp

```diff
@@ -895,4 +895,13 @@ RLM_API void realm_sync_session_wait_for_upload_completion(realm_sync_session_t*
         };
     (*session)->wait_for_upload_completion(std::move(cb));
 }
+
+RLM_API void realm_sync_session_handle_error_for_testing(realm_sync_session_t* session, realm_sync_error_t* error)
+{
+    REALM_ASSERT(session);
+    REALM_ASSERT(error);
+    auto err = std::make_error_code(std::errc{error->error_code.value});
```

Comment:
We can't use this form, because not all error codes are `std::errc`. Most of them are actually sync-specific. This function needs to construct an `std::error_code` from `realm_sync_error_code_t` by reversing the logic here: https://github.com/realm/realm-core/blob/336cae9536938fa65ab8fe040836f31c22db40fa/src/realm/object-store/c_api/sync.cpp#L188-L225